Johnson_Algorithm/Johnson_Algorithm.c

```diff
@@ -0,0 +1,386 @@
+    //printGraph(graph);
+    
+    //Matrix to store distance of every vertex with every other vertex
+    int distance[V][V];
+    //Running Dijsktra for every vertex as source
+    for(int i=0; i<V; i++)
+    {
+        int __dist[V];
+        
+        // Calculating shortest distance from source i to every other vertices using Dijsktra
+        int* dist_final = Dijsktra(graph, V, E, i, __dist);
+        
+        // Update the final matrix distance
+        for(int j=0; j<V; j++)
+        {
+            distance[i][j] = dist_final[j];
+        }
+    }
+    
+    printf("Final Result\n");
+    printf("(Value 2147483648 indicates vertex v is not reachable from vertex u)\n\n");
+    for(int i=0; i<V; i++)
+    {
+        for(int j=0; j<V; j++)
+        {
+            if(distance[i][j]==INT_MAX)
+            {
+                distance[i][j] = INT_MAX;
+                // To indicate that j vertex is not reachable from i vertex
+                // No Edge from i to j
+            }
+            printf("Distance from %d to %d is :%d\n",i, j, distance[i][j]);
+        }
+        printf("\n");
+    }
+}
+
+/*
+      INPUT:
+      
+      7 9
+      0 1 -1
+      0 2 0
+      3 0 12
+      3 2 5
+      2 5 -6
+      5 4 3
+      4 1 4
+      6 5 8
+      1 6 -7
+      
+      OUTPUT:
+      
+      Final Result
+(Value -2147483648 indicates vertex v is not reachable from vertex u)
+
+Distance from 0 to 0 is :0
+Distance from 0 to 1 is :0
+Distance from 0 to 2 is :0
+Distance from 0 to 3 is :-2147483648
+Distance from 0 to 4 is :0
+Distance from 0 to 5 is :0
+Distance from 0 to 6 is :0
+
+Distance from 1 to 0 is :-2147483648
+Distance from 1 to 1 is :0
+Distance from 1 to 2 is :-2147483648
+Distance from 1 to 3 is :-2147483648
+Distance from 1 to 4 is :6
+Distance from 1 to 5 is :6
+Distance from 1 to 6 is :0
+
+Distance from 2 to 0 is :-2147483648
+Distance from 2 to 1 is :2
+Distance from 2 to 2 is :0
+Distance from 2 to 3 is :-2147483648
+Distance from 2 to 4 is :0
+Distance from 2 to 5 is :0
+Distance from 2 to 6 is :2
+
+Distance from 3 to 0 is :12
+Distance from 3 to 1 is :7
+Distance from 3 to 2 is :5
+Distance from 3 to 3 is :0
+Distance from 3 to 4 is :5
+Distance from 3 to 5 is :5
+Distance from 3 to 6 is :7
+
+Distance from 4 to 0 is :-2147483648
+Distance from 4 to 1 is :2
+Distance from 4 to 2 is :-2147483648
+Distance from 4 to 3 is :-2147483648
+Distance from 4 to 4 is :0
+Distance from 4 to 5 is :8
+Distance from 4 to 6 is :2
+
+Distance from 5 to 0 is :-2147483648
+Distance from 5 to 1 is :2
+Distance from 5 to 2 is :-2147483648
+Distance from 5 to 3 is :-2147483648
+Distance from 5 to 4 is :0
+Distance from 5 to 5 is :0
+Distance from 5 to 6 is :2
+
+Distance from 6 to 0 is :-2147483648
+Distance from 6 to 1 is :8
+Distance from 6 to 2 is :-2147483648
+Distance from 6 to 3 is :-2147483648
+Distance from 6 to 4 is :6
+Distance from 6 to 5 is :6
+Distance from 6 to 6 is :0
+
+        
+ */
+
+
+
+
```

Comment:
There should be a single new line at the end of file.

Reply:
Done @jainaman224